pkg/build/controller/buildconfig_controller.go

```diff
@@ -35,14 +36,21 @@ func IsFatal(err error) bool {
 
 type BuildConfigController struct {
 	BuildConfigInstantiator buildclient.BuildConfigInstantiator
-
+	BuildConfigGetter       buildclient.BuildConfigGetter
+	BuildLister             buildclient.BuildLister
+	BuildDeleter            buildclient.BuildDeleter
 	// recorder is used to record events.
 	Recorder record.EventRecorder
 }
 
 func (c *BuildConfigController) HandleBuildConfig(bc *buildapi.BuildConfig) error {
 	glog.V(4).Infof("Handling BuildConfig %s/%s", bc.Namespace, bc.Name)
 
+	glog.V(4).Infof("Handling Build Pruning: %v/%v", bc.Namespace, bc.Name)
+	if err := buildutil.HandleBuildPruning(bc.Name, bc.Namespace, c.BuildLister, c.BuildConfigGetter, c.BuildDeleter); err != nil {
+		return err
+	}
```

Comment:
1) why isn't this returning the same wrapper error message as HandleBuildCompletion does when it has an error pruning?

2) i think you should proceed through the rest of the HandleBuildConfig logic and return an aggregated error object at the end, rather than quitting here if something goes wrong.  Even if pruning fails, we still want to kick off the new build.

(Unless the HandleBuildConfig event gets retried when an error is encountered?)